src/ast/passes/config_analyser.cpp

```diff
@@ -62,6 +63,13 @@ void ConfigAnalyser::visit(AssignConfigVarStatement &assignment)
     assignment.addWarning()
         << assignment.var << " has been renamed, please use " << var;
   }
+
+  // Need this in a fairly early pass to make sure we don't get warnings
+  // in later passes.
+  if (!bpftrace_.config_->log_warnings) {
+    bpftrace_.helper_check_level_ = 0;
+    DISABLE_LOG(WARNING);
+  }
```

Comment:
I think the pass mutating the global log state is a bit of a no-no. Can we just put this into `main` after the passes have been run, but prior to actually running the pipelines? It should essentially be in the same place as the `--no-warnings` handler.

Reply:
> but ideally we would have a model where those are pushed to the individual probes

Do you mean something like probe attributes? 

The specifics on this change are that there were runtime warnings/errors for map functions in scripts where these errors were somewhat expected. So not really "probe" specific in that case. Maybe another solution might be named arguments for some of these map functions e.g. `delete(@a, 1, warnings=false)`.